src/states/states.ts

```diff
@@ -0,0 +1,381 @@
+
+  constructor(name: string, supplies: any) {
+    super(name);
+    this.type = 'SupplyList';
+    this.supplies = supplies;
+  }
+
+  toJSON() {
+    return {
+      ...super.toJSON(),
+      supplies: this.supplies
+    };
+  }
+}
+
+class VitalSignState extends BaseState {
+  vitalSign: any;
+  unit: string | null;
+
+  constructor(name: string, vitalSign: any, unit: string | null) {
+    super(name);
+    this.type = 'VitalSign';
+    this.vitalSign = vitalSign;
+    this.unit = unit;
+  }
+
+  toJSON() {
+    return {
+      ...super.toJSON(),
+      vital_sign: this.vitalSign,
+      unit: this.unit
+    };
+  }
+}
+
+class ObservationState extends BaseState {
+  category: string | null;
+  unit: string | null;
+  codes: any;
+
+  constructor(name: string, category: string | null, unit: string | null, codes: any) {
+    super(name);
+    this.type = 'Observation';
+    this.category = category;
+    this.unit = unit;
+    this.codes = codes;
+  }
+
+  toJSON() {
+    return {
+      ...super.toJSON(),
+      category: this.category,
+      unit: this.unit,
+      codes: this.codes,
+      exact: {
+        quantity: 1
+      }
+    };
+  }
+}
+
+class MultiObservationState extends BaseState {
+  category: string | null;
+  numberOfObservations: number | null;
+  codes: any;
+
+  constructor(name: string, category: string | null, numberOfObservations: number | null, codes: any) {
+    super(name);
+    this.type = 'MultiObservation';
+    this.category = category;
+    this.numberOfObservations = numberOfObservations;
+    this.codes = codes;
+  }
+
+  toJSON() {
+    return {
+      ...super.toJSON(),
+      category: this.category,
+      number_of_observations: this.numberOfObservations,
+      codes: this.codes
+    };
+  }
+}
+
+class DiagnosticReportState extends BaseState {
+  numberOfObservations: number | null;
+  codes: any;
+
+  constructor(name: string, numberOfObservations: number | null, codes: any) {
+    super(name);
+    this.type = 'DiagnosticReport';
+    this.numberOfObservations = numberOfObservations;
+    this.codes = codes;
+  }
+
+  toJSON() {
+    return {
+      ...super.toJSON(),
+      number_of_observations: this.numberOfObservations,
+      codes: this.codes
+    };
+  }
+}
+
+class SymptomState extends BaseState {
+  symptom: any;
+  cause: any;
+  probability: any;
+
+  constructor(name: string, symptom: any, cause: any, probability: any) {
+    super(name);
+    this.type = 'Symptom';
+    this.symptom = symptom;
+    this.cause = cause;
+    this.probability = probability;
+  }
+
+  toJSON() {
+    return {
+      ...super.toJSON(),
+      symptom: this.symptom,
+      cause: this.cause,
+      probability: this.probability
+    };
+  }
+}
+class DeathState extends BaseState {
+  constructor(name: string) {
+    super(name);
+    this.type = 'Death';
+  }
+}
+class TerminalState extends BaseState {
+  constructor(name: string) {
+    super(name);
+    this.type = 'Terminal';
+  }
+}
+
+module.exports = {
```

Comment:
I wonder if we should be simply slapping `exports` in front of `class` on all of the classes instead of using the `module.exports` construct.

Thoughts on this @mgramigna.

Reply:
yes. we should be using `export class ____` instead, now that we get that nice syntax for free with Typescript.

module.exports? never heard of it